src/explorers/AAPS.jl

```diff
@@ -48,18 +48,45 @@ function adapt_explorer(explorer::AAPS, reduced_recorders, current_pt, new_tempe
     estimated_target_std_deviations = adapt_preconditioner(explorer.preconditioner, reduced_recorders)
     # TODO: adapt step_size and K
     return AAPS(
-        explorer.step_size, explorer.K, explorer.default_autodiff_backend,
+        explorer.step_size_ref, explorer.K, explorer.default_autodiff_backend,
         explorer.preconditioner, estimated_target_std_deviations
     )
 end
 
+# uses the autoMALA internal `auto_step_size` to find a step size
+function find_reasonable_step_size(
+    old_step_size::Real,
+    replica, 
+    target_log_potential, 
+    state::AbstractVector;
+    lower_bound = -3.0, # exp(-3) ~ 5%
+    upper_bound = 0.0
+    )
+    recorders      = replica.recorders
+    dim            = length(state)
+    temp_position  = get_buffer(recorders.buffers, :aaps_fwd_position_buffer, dim)
+    temp_velocity  = get_buffer(recorders.buffers, :aaps_fwd_velocity_buffer, dim)
+    temp_precond   = get_buffer(recorders.buffers, :aaps_diag_precond, dim)
+    temp_position .= state
+    randn!(replica.rng, temp_velocity)
+    fill!(temp_precond, one(eltype(temp_precond)))
+    exponent = auto_step_size(
+        target_log_potential, temp_precond, temp_position, temp_velocity, 
+        recorders, replica.chain, old_step_size, lower_bound, upper_bound)
+    return old_step_size * (2.0^exponent)
+end
+
 #=
 Extract info common to all types of target and perform a step!()
 =#
 function _extract_commons_and_run!(explorer::AAPS, replica, shared, log_potential, state::AbstractVector)
     log_potential_autodiff = ADgradient(
         explorer.default_autodiff_backend, log_potential, replica.recorders.buffers
-    )      
+    )
+    if shared.iterators.round == 1 && shared.iterators.scan == 1
+        explorer.step_size_ref[] = find_reasonable_step_size(
```

Comment:
@alexandrebouchard I wanted your opinion on this. Right now it seems that this adaptation would occur across all chains, which seems bad. Where could I place this initialization if not here? The issue is that I need the ADgradient to exist and this is where it is first created.

Reply:
I guess I could skip the adaptation if `n_chains > 1` for now to avoid having all chains writing to the same ref[].